pennylane/_qubit_device.py

```diff
@@ -836,16 +844,19 @@ def adjoint_jacobian(self, tape):
             if not hasattr(m.obs, "base_name"):
                 m.obs.base_name = None  # This is needed for when the observable is a tensor product
 
-        # Perform the forward pass.
-        # Consider using caching and calling lower-level functionality. We just need the device to
-        # be in the post-forward pass state.
-        # https://github.com/PennyLaneAI/pennylane/pull/1032/files#r563441040
-        self.reset()
-        self.execute(tape)
+        if starting_state is not None:
+            ket = self._reshape(starting_state, [2] * self.num_wires)
```

Comment:
I wanted to make it so that someone could do

```
state = dev.state
dev.adjoint_jacobian(qtape, starting_state=state)
```

`dev.state` is flattened, so I added this reshape to accommodate doing the above.

Reply:
doing so.

Reply:
Actually, I just realized that reshaping is an incredibly efficient operation because it is a different view of the same data.  And this decreases readability slightly.  So I'm always going to reshape starting state.

Reply:
Just ran some example timing and had to laugh at how negligible the cost of reshaping are! :D Nice!